### PR TITLE
Fix links on plugins docs page.

### PR DESCRIPTION
**Pull request by phiggins**
Signed-off-by: Pete Higgins <pete@peterhiggins.org>

## Description

Fix some broken links on the plugins docs page.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

Aha! Link: https://chef.aha.io/features/SH-406

### CODE REVIEW

File: docs-chef-io/content/inspec/plugins.md [hunk 1]
```diff
@@ -78,7 +77,7 @@ inspec plugin install --source https://my.private.server inspec-private-plugin
 ### Chef InSpec Plugins
 
 For details on how to author a Chef InSpec Plugin, see the
-[developer documentation](https://github.com/inspec/inspec/blob/master/docs/dev/plugins.md)
+[developer documentation](https://github.com/inspec/inspec/blob/master/dev-docs/plugins.md)
```
Comment (phiggins) [inline docs-chef-io/content/inspec/plugins.md:80]: This link was broken, it appears to be referring to [this document](https://github.com/inspec/inspec/blob/master/dev-docs/plugins.md) so I updated the link.

File: docs-chef-io/content/inspec/plugins.md [hunk 2]
```diff
@@ -60,8 +60,7 @@ inspec plugin install inspec-some-plugin
 inspec plugin install train-some-plugin
 ```
 
-For more details on what the `plugin` command can do, see the [online help](/inspec/cli/#plugin),
-or run `inspec plugin help`.
+For more details on what the `plugin` command can do, run `inspec plugin help`.
```
Comment (phiggins) [inline docs-chef-io/content/inspec/plugins.md:63]: This linked to [an existing document](https://docs.chef.io/inspec/cli/), but that does not contain any references to the `inspec plugin` command so I removed the link.

Comment (mjingle) [reply docs-chef-io/content/inspec/plugins.md:63]: This sentence will probably need to be re-worded then.

Comment (james-stocks) [reply docs-chef-io/content/inspec/plugins.md:63]: As a follow-up to this PR we should keep an issue open for the fact that https://docs.chef.io/inspec/cli/ does not mention the `plugin` or `init` commands, I'll raise that now.

Comment (james-stocks) [reply docs-chef-io/content/inspec/plugins.md:63]: https://github.com/inspec/inspec/issues/5291

Comment (mjingle) [reply docs-chef-io/content/inspec/plugins.md:63]: @phiggins Scratch that about more rewording. I had more coffee. 😅 Your rewording should be good as is.
@james-stocks Thanks in advance for filing that issue!

**Review by mjingle — APPROVED** [review]
Thank you so much for catching this and updating this page, @phiggins!

**Review by james-stocks — APPROVED** [review]
Thanks very much Pete!

**Pull request merged**